database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostListViewModel.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  Copyright (c) 2021 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import Firebase
+
+class PostListViewModel: ObservableObject {
+  @Published var isLoading = false
+  @Published var alert = false
+  @Published var alertMessage = ""
+  @Published var posts: [PostViewModel] = []
+
+  // setup instance of FIRDatabaseReference for reading and writing data
+  private var ref = Database.root
+  private var refHandle: DatabaseHandle?
+
+  // function called when error occurs
+  func showAlertMessage(message: String) {
+    alertMessage = message
+    alert.toggle()
+  }
+
+  func getCurrentUserID() -> String? {
+    return Auth.auth().currentUser?.uid
+  }
+
+  // function called when user is making a new post
+  func didTapPostButton(title: String, body: String) {
+    // check if both title and body are completed
+    if title.isEmpty || body.isEmpty {
+      showAlertMessage(message: "Neither title nor body can be empty.")
+      return
+    }
+
+    // begin loading animation
+    withAnimation {
```

Comment:
Though it works, it violates MVVM as ViewModel should be UI agnostic and work with any View implementation. I would suggest removing the animation at first and then think of a proper way to implement it when have time.
I don't think it's a blocker for the current PR, but it should be eventually addressed.

Reply:
I see, I will remove the animation step from file for now and see if there is a way I can implement it without violating the MVVM model.